pkg/controller/installation/core_controller.go

```diff
@@ -1067,11 +1067,10 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	}
 	var managerInternalTLSSecret certificatemanagement.KeyPairInterface
 	if instance.Spec.Variant == operator.TigeraSecureEnterprise {
-		dnsNames := append(dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain), render.ManagerServiceIP)
-		managerInternalTLSSecret, err = certificateManager.GetOrCreateKeyPair(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace(), dnsNames)
+		managerInternalTLSSecret, err = certificateManager.GetKeyPair(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace())
```

Comment:
With this change, won't `managerInternalTLSSecret` for a new installation will be nil (since the manager controller needs to wait for the core controller to setup networking)? It looks like the Objects function can handle nil key pairs, but it might still be better to not add this as an option if it's nil.

You may also want to investigate if you even need this tls secret in the core controller. I don't think anything here uses voltron anymore, it might just be left over from before we split kube controllers into two components (i.e. the es kube controllers).

Reply:
Very insightful, I had not thought of that.
- Now mounting the ca-bundle in the kubecontrollers, rather than the internal-manager-tls secret.
- We still need to watch the secret in order to add it to the bundle in the calico-system namespace.